Plugin/Catalog/Block/Toolbar.php

```diff
@@ -186,4 +197,35 @@ private function manipulateSQL(FullTextCollection $collection, array $ids)
         $zendExpression = new \Zend_Db_Expr('FIELD(e.entity_id,' . implode(',', $ids) . ') DESC');
         $select->order($zendExpression);
     }
+
+    /**
+     * @param CategoryMerchandisingResult $result
+     * @return array
+     */
+    private function getSortedIds(CategoryMerchandisingResult $result)
+    {
+        $productIds = [];
+        try {
+            foreach ($result->getResultSet() as $item) {
+                if ($item->getProductId() && is_numeric($item->getProductId())) {
+                    $productIds[] = $item->getProductId();
+                }
+            }
+        } catch (\Exception $e) {
+            $this->logger->exception($e);
+        }
+
+        return $productIds;
+    }
+
+    /**
+     * @param FullTextCollection $collection
+     * @param string $trackCode
+     */
+    private function addTrackParamToProduct(FullTextCollection $collection, $trackCode)
```

Comment:
Let's type hint this as `Magento\Catalog\Model\ResourceModel\Product\Collection`